File: cpp/Utility/OrderAttribute.c
// Type: Utility.OrderAttribute
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Utility/OrderAttribute.cs
// Prior real C# source (older reference): Assets/Scripts/Utility/CSV/OrderAttribute.cs
// --------------------------------

// Utility.OrderAttribute$$get_Order
// il2cpp: int32_t Utility_OrderAttribute__get_Order (Utility_OrderAttribute_o* __this, const MethodInfo* method);
// 0x448eeb0

int32_t Utility_OrderAttribute__get_Order(Utility_OrderAttribute_o *__this,MethodInfo *method)

{
  return (__this->fields).order;
}


// Utility.OrderAttribute$$.ctor
// il2cpp: void Utility_OrderAttribute___ctor (Utility_OrderAttribute_o* __this, int32_t order, const MethodInfo* method);
// 0x448eec0

void Utility_OrderAttribute___ctor(Utility_OrderAttribute_o *__this,int32_t order,MethodInfo *method)

{
  System_Attribute___ctor((System_Attribute_o *)__this,(MethodInfo *)0x0);
  (__this->fields).order = order;
  return;
}

File: csharp_body_deref/Scripts/Utility/OrderAttribute.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Utility.OrderAttribute
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Utility/OrderAttribute.c
// Prior real C# source (older reference): Assets/Scripts/Utility/CSV/OrderAttribute.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System;
using Il2CppDummyDll;

namespace Utility;

[Token(Token = "0x2000732")]
[AttributeUsage(AttributeTargets.Field, Inherited = true, AllowMultiple = false)]
internal class OrderAttribute : Attribute
{
	[Token(Token = "0x4002293")]
	[FieldOffset(Offset = "0x10")]
	private readonly int order;

	[Token(Token = "0x17000DBA")]
	public int Order
	{
		[Token(Token = "0x60046B3")]
		[Address(RVA = "0x438EEB0", Offset = "0x438EEB0", VA = "0x438EEB0")]
		get
		{
			return default(int);
		}
	}

	[Token(Token = "0x60046B4")]
	[Address(RVA = "0x438EEC0", Offset = "0x438EEC0", VA = "0x438EEC0")]
	public OrderAttribute(int order)
	{
	}
}
